src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -544,7 +549,7 @@ private static MethodBodyStatement CreatePropertyNullCheckStatement(JsonProperty
                 };
             }
 
-            if (!jsonPropertySerialization.IsRequired &&
+            if ((!jsonPropertySerialization.IsRequired || TypeFactory.IsReadOnlyMemory(serializedType!)) && // even if ReadOnlyMemory is required we leave the list empty if the payload doesn't have it
```

Comment:
`serializedType` is not guaranteed to be not null here. We should check for null explicitly.